Add Ability to Split/Merge Steps in Recipe Editor View

birthdayattack opened the pull request:
Closes #3802 

This adds buttons for: 
- Splitting step into individual steps based on newline
- Merging all steps into one step

into the recipe editor view. This reuses the logic already present in the recipe importer view.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/TandoorRecipes/recipes?pullRequest=3890) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/TandoorRecipes/recipes?pullRequest=3890) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/TandoorRecipes/recipes?pullRequest=3890) it.</sub>

vabene1111 commented:
thanks for the PR. Since the import Recipe and the Recipe in the Editor share the mostly the same datastructure can you please verify if the same functions can be used to perform the operations because duplicating the functions is not good. 

Best would be in some utility file/class.

birthdayattack commented:
Good point. I've pulled out the logic into a separate step_utils file.

The Step and SourceImportStep interfaces are similar, but separate, and those files are autogenerated by OpenAPI. So I added a StepLike interface into the same step_utils, to act as a generic Step object. 

I also went ahead and updated the merge code to use map and flatMap instead of the existing forEach. I noticed a newline gets appended to the merge instruction each time you click the button.

vabene1111 commented:
perfect, thats exactly what I preferred as a solution.  Will get this merged with the next update

vabene1111 commented:
hi, for some reason the GitHub diff viewer became really bad and its hard for me to see where the changes were made. Would you mind merging your branch locally and updating based on the changes I have made to the RecipeEditor so I can merge more easily?

birthdayattack commented:
hm, looks like the diff view is still being difficult. I did merge the changes locally. Let me know if there are any formatting reasons why the diff might be confused. I'm using the formatting extensions [mentioned in the docs for VSCode.](https://docs.tandoor.dev/contribute/vscode/)
